source/extensions/filters/network/thrift_proxy/router/router_impl.h

```diff
@@ -52,11 +52,27 @@ class MethodNameRouteEntryImpl : public RouteEntryImplBase {
 
   const std::string& methodName() const { return method_name_; }
 
-  // RoutEntryImplBase
+  // RouteEntryImplBase
   RouteConstSharedPtr matches(const MessageMetadata& metadata) const override;
 
 private:
   const std::string method_name_;
+  bool invert_;
+};
+
+class ServiceNameRouteEntryImpl : public RouteEntryImplBase {
+public:
+  ServiceNameRouteEntryImpl(
+      const envoy::config::filter::network::thrift_proxy::v2alpha1::Route& route);
+
+  const std::string& serviceName() const { return service_name_; }
+
+  // RouteEntryImplBase
+  RouteConstSharedPtr matches(const MessageMetadata& metadata) const override;
+
+private:
+  std::string service_name_;
+  bool invert_;
```

Comment:
nit: both of these should be const. That said, it makes the constructor tricky so, I think I'm ok with leaving service_name_ non-const.